pkg/volume/glusterfs/glusterfs.go

```diff
@@ -403,15 +433,15 @@ func (b *glusterfsMounter) setUpAtInternal(dir string) error {
 
 }
 
-func getVolumeSource(spec *volume.Spec) (*v1.GlusterfsVolumeSource, bool, error) {
+func getVolumeSource(spec *volume.Spec) (string, string, bool, error) {
 	if spec.Volume != nil && spec.Volume.Glusterfs != nil {
-		return spec.Volume.Glusterfs, spec.Volume.Glusterfs.ReadOnly, nil
+		return spec.Volume.Glusterfs.EndpointsName, spec.Volume.Glusterfs.Path, spec.Volume.Glusterfs.ReadOnly, nil
+
 	} else if spec.PersistentVolume != nil &&
 		spec.PersistentVolume.Spec.Glusterfs != nil {
-		return spec.PersistentVolume.Spec.Glusterfs, spec.ReadOnly, nil
+		return spec.PersistentVolume.Spec.Glusterfs.EndpointsName, spec.PersistentVolume.Spec.Glusterfs.Path, spec.PersistentVolume.Spec.Glusterfs.ReadOnly, nil
```

Comment:
don't change the readonly behavior as part of this PR. please open that separate as a bugfix and get close eyes on the change from @kubernetes/sig-storage-pr-reviews ... need to think through whether that is compatible or not